### ISSUE TITLE
SimplifyBooleanExpression: A false negative about ternary operator

### ISSUE THREAD

**Issue by Zustin** [checkstyle/checkstyle#11015]
I have read check documentation: https://checkstyle.sourceforge.io/config_coding.html#SimplifyBooleanExpression
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Test.java
/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <property name="severity" value="error"/>
  <property name="fileExtensions" value="java, properties, xml"/>
  <module name="TreeWalker">    
    <module name="SimplifyBooleanExpression"/>
  </module>
</module>

/var/tmp $ cat Test.java
public class Test {
    public boolean fun() {
        return false;
    }
    public boolean foo() {
        boolean tag = fun() ? true : false;  // should report a warning here
        // boolean tag = func() == true; // can be detected
        return tag;
    }
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-9.2-all.jar -c ./config.xml Test.java

#[[PLACE YOUR OUTPUT HERE]]
Starting audit...
Audit done.
```
---
**Describe what you expect in detail.**
---
I have just found a false negative about the rule SimplifyBooleanExpression. Please refer to the below simplfied case. Here, I think `fun() ? true : false`  at line 6 is similar to `func() == true`. They all can be simplified into `func()`.

**Comment by Vyom-Yadav** [comment]
@Zustin Please follow the issue template, it will help us figure out what you expect.

**Comment by Zustin** [comment]
@Vyom-Yadav Hi Yadav, thanks for your kind reply. I have updated the issue comment. Please check it again. Thanks. :smile:

**Comment by strkkk** [comment]
@Zustin it is not quite what is required. Please change config to only check that you need, output should not have any irrelevant violations, see https://github.com/checkstyle/checkstyle/issues/11001 as an example

**Comment by Vyom-Yadav** [comment]
@Zustin Hi, Thanks for the update but it's not quite correct, config.xml and CLI output should only include the check about which we are talking right now also please include config.xml as text file and not as a link

**Comment by Zustin** [comment]
Hi @strkkk , I have updated my issue. Please refer to it, thanks for your kindest help.

**Comment by Vyom-Yadav** [comment]
@strkkk Was able to reproduce and verify this, would be willing to work on this once approved

**Comment by strkkk** [comment]
@Vyom-Yadav issue is approved, this case clearly can be detected.
Please note that reversed case `func() ? false : true` should be also checked

P.S. I deleted irrelevant comments

**Comment by romani** [comment]
fix is merged
